File: server/core/api/survey.go
package api

import "letstalk/server/data"

type SurveyOption struct {
	Key  data.SurveyOptionKey `json:"key" binding:"required"`
	Text string          `json:"text" binding:"required"`
}

type SurveyQuestion struct {
	Key     data.SurveyQuestionKey `json:"key" binding:"required"`
	Prompt  string            `json:"prompt" binding:"required"`
	Options []SurveyOption    `json:"options" binding:"required"`
}

type Survey struct {
	Version   data.SurveyVersion   `json:"version" binding:"required"`
	Questions []SurveyQuestion     `json:"questions" binding:"required"`
	Responses data.SurveyResponses `json:"responses" binding:"required"`
}

File: letstalk/src/models/survey.ts
import Immutable from 'immutable';

export interface SurveyOption {
  readonly key: string
  readonly text: string
}

export interface SurveyQuestion {
  readonly key: string
  readonly prompt: string
  readonly options: Immutable.List<SurveyOption>
}

export type SurveyQuestions =  Immutable.List<SurveyQuestion>
export type SurveyResponses = Immutable.Map<string, string>;

export interface Survey {
  readonly group: string
  readonly version: number
  readonly questions: SurveyQuestions
  readonly responses?: SurveyResponses
}
